.github/actions/javascript/proposalPoliceComment/proposalPoliceComment.ts

```diff
@@ -0,0 +1,106 @@
+import {setFailed} from '@actions/core';
+import {context} from '@actions/github';
+import type {IssueCommentCreatedEvent, IssueCommentEditedEvent, IssueCommentEvent} from '@octokit/webhooks-types';
+import CONST from '@github/libs/CONST';
+import GithubUtils from '@github/libs/GithubUtils';
+import OpenAIUtils from '@github/libs/OpenAIUtils';
+
+function isCommentCreatedOrEditedEvent(payload: IssueCommentEvent): payload is IssueCommentCreatedEvent | IssueCommentEditedEvent {
+    return payload.action === CONST.ACTIONS.CREATED || payload.action === CONST.ACTIONS.EDIT;
+}
+
+function isCommentCreatedEvent(payload: IssueCommentEvent): payload is IssueCommentCreatedEvent {
+    return payload.action === CONST.ACTIONS.CREATED;
+}
+
+// Main function to process the workflow event
+async function run() {
+    // Verify this is running for an expected webhook event
+    if (context.eventName !== CONST.EVENTS.ISSUE_COMMENT) {
+        throw new Error('ProposalPolice™ only supports the issue_comment webhook event');
+    }
+
+    const payload = context.payload as IssueCommentEvent;
+
+    // check if the issue is open and the has labels
+    if (payload.issue?.state !== 'open' && !payload.issue?.labels.some((issueLabel: {name: string}) => issueLabel.name === CONST.LABELS.HELP_WANTED)) {
+        return;
+    }
+
+    // Verify that the comment is not empty
+    if (!payload.comment?.body.trim()) {
```

Comment:
Can we also check if the comment begins with `# Proposal`? So we can only check the comments that contain a proposal, otherwise I think we can trigger the OpenAI API in all the comments

Reply:
@marcochavezf This is handled by the AI assistant on both new comments which checks if a proposal is following the template and also on proposal edits to check for significant changes.

> Can we also check if the comment begins with `# Proposal`?

This wouldn't be a good idea because currently for if AI assistant detects a proposal has significant changes, it will add a line like: `Edited by proposal-police: This proposal was edited at 2024-06-22 22:28:40 UTC.` at the top of the proposal which would break the logic proposed by you (see example in https://github.com/ikevin127/expensify-proposal-testing/issues/1#issuecomment-2184213860).

##

For details like this we decided to rely on the AI assistant via [instructions](https://pastebin.com/raw/DVx9QJpy) because we could have users add different proposal template markdown (or just text) which would still be valid like: `Proposal`, `# Proposal`, `## Proposal`.

If we take a look at the given AI [instructions](https://pastebin.com/raw/DVx9QJpy) you'll notice that we made sure to instruct for variations like these and also for if any other things are added above / below the proposal like the `Edited by proposal-police...` AI would look at the whole comment and if the proposal follows the template (+/- acceptable markdown / text variations), the AI would know about this and only respond if the proposal template is actually not followed (different RCA / solution / alt solution headings, ex: https://github.com/ikevin127/expensify-proposal-testing/issues/1#issuecomment-1934900789).

Reply:
If we want to optimize this and not run the AI on every new comment, I guess we can be very strict and check if new issue comment contains `# Proposal` and only if true we run the AI, otherwise don't run at all ? But this would have the following edge case:
- if user posts proposal using say `## Proposal` or plain text `Proposal` and AI won't run on it -> nothing will happen (AI won't post the "you did not follow the proposal template" comment)

Reply:
What if we just check if the comment contains the word proposal (case-insensitive)? The problem is that we generate a considerable number of issues per month, and each issue could have multiple comments where it's not necessary to send an OpenAI request

Reply:
Ya, I think that can definitely be done in order to reduce the amount of times we call the AI, therefore reduce costs.
Will get on this tomorrow and make the change.

Reply:
@marcochavezf Done, synced w/ main and added the discussed change to make sure that the comment body includes the keyword `Proposal` case-sensitive.